sdk/nodejs/appservice/zMixins.ts

```diff
@@ -312,15 +370,56 @@ function redirectConsoleOutput<C extends Context<R>, E, R extends Result>(callba
     };
 }
 
+/**
+ * Interface to define individual Azure Functions. A function is not deployable by itself: instead, it must be passed
+ * to MultiFunctionApp to be deployed as a part of the Azure Function App.
+ */
+export interface Function {
+    /**
+     * Azure Function name.
+     */
+    name: string;
+
+    /**
+     * An array of function binding definitions.
+     */
+    bindings: pulumi.Input<BindingDefinition[]>;
+
+    /**
+     * Function callback.
+     */
+    callback: CallbackArgs<Context<any>, any, any>;
+
+    /**
+     * Application settings required by the function.
+     */
+    appSettings?: pulumi.Input<{ [key: string]: any }>;
+}
 
 /**
-  * A CallbackFunctionApp is a special type of azure.appservice.FunctionApp that can be created out
-  * of an actual JavaScript function instance.  The function instance will be analyzed and packaged
-  * up (including dependencies) into a form that can be used by AWS Lambda.  See
-  * https://github.com/pulumi/docs/blob/master/reference/serializing-functions.md for additional
-  * details on this process.
+ * Arguments to create a Function App component with multiple callback functions in it.
+ */
+export interface MultiCallbackFunctionAppArgs extends FunctionAppArgsBase {
+    /**
+     * The functions to deploy as parts of this application. At least 1 function is required.
+     */
+    functions: Function[];
+};
+
+/**
+ * Arguments to create a Function App component and deploy the specified raw archive package.
+ */
+export interface ArchiveFunctionAppArgs extends FunctionAppArgsBase {
+    /**
+     * The deployment package of a Function App to deploy as-is.
+     */
+    archive: pulumi.Input<pulumi.asset.Archive>;
+};
+
+/**
+ * A composite resouce which creates a Function App and all required resources that the app depends on.
  */
-export class CallbackFunctionApp<C extends Context<R>, E, R extends Result> extends FunctionApp {
+class CombinedFunctionApp extends FunctionApp {

```

Comment:
should this be abstract?  we woudln't expect this to be instnatiated directly right?

Reply:
n/m.  i see what you're doing.